### ISSUE TITLE
Input variable to exit if failed

### ISSUE THREAD

**Issue by LecrisUT** [beakerlib/beakerlib#175]
I am considering running some tests with `beakerlib` inside a spec file. The issue is the tests will always pass because it has `exit 0`. I could add a check for `BEAKERLIB_DIR`, but it would be nice to have an alternative to return an exit code if failed.

**Comment by sopos** [comment]
Right. There's no exit code as beakerlib basically cannot control how the test itself exits. For that purpose it generates the `$BEAKERLIB_DIR/TestResults` file which is bash-sourceable. Inside it you can find `TESTRESULT_RESULT_ECODE` which reflects the test state. If the `TESTRESULT_STATE=="complete" && TESTRESULT_RESULT_ECODE == "0"`, you can consider the test as passed.
May be some simple wrapper as a dummy executor could help here to translate it to the real exit code.

**Comment by LecrisUT** [comment]
I have tried to write something as:
```
source $(pwd)/tests_results_${mpi:-serial}/TestResults
[ "${TESTRESULT_RESULT_STRING}" == "FAIL" ] && exit 1
```
But this didn't actually pass: https://src.fedoraproject.org/rpms/cp2k/pull-request/6

It is rather weird

**Comment by sopos** [comment]
No idea here. It would be good to consul someone more familiar with the build process.
